src/database/sync.database.ts

```diff
@@ -1,96 +1,80 @@
-import {
-  DeleteRowData,
-  ImportQueue,
-  UnsyncedBlock,
-  QueueState,
-} from '../types/cassandra.types';
+import * as R from "rambda";
+import Fluture, { fork, parallel } from "fluture/index.js";
+import PriorityQueue from "../utility/priority.queue";
+import pWaitFor from "p-wait-for";
+import { DataItemJson } from "arweave-bundles";
+import Gauge from "gauge";
+import GaugeThemes from "gauge/themes";
+import { config } from "dotenv";
+import { types as CassandraTypes } from "cassandra-driver";
+import { KEYSPACE, POLLTIME_DELAY_SECONDS } from "../constants";
+import { MAX_TX_PER_BLOCK } from "./constants.database";
+import { log } from "../utility/log.utility";
+import { ansBundles } from "../utility/ans.utility";
+import { mkdir } from "../utility/file.utility";
+import { getDataFromChunks, getHashList, getNodeInfo } from "../query/node.query";
+import { fetchBlockByHash, getBlock as queryGetBlock } from "../query/block.query";
+import { getTransaction, getTxOffset } from "../query/transaction.query";
+import { QueueState, UnsyncedBlock } from "../types/cassandra.types";
 import {
   cassandraClient,
   getMaxHeightBlock,
   makeBlockImportQuery,
   makeTxImportQuery,
   toLong,
-} from './cassandra.database';
-import { blockMapper, blockHeightToHashMapper } from './mapper.database';
-import * as Dr from './doctor.database';
-import { cacheANSEntries } from '../caching/ans.entry.caching';
+} from "./cassandra.database";
+import * as Dr from "./doctor.database";
+import { cacheANSEntries } from "../caching/ans.entry.caching";
 
-process.env.NODE_ENV !== 'test' && config();
-mkdir('cache');
+process.env.NODE_ENV !== "test" && config();
+mkdir("cache");
 
 const trackerTheme = GaugeThemes.newTheme(
-  GaugeThemes({
-    hasUnicode: false,
-    hasColor: true,
-  })
+    GaugeThemes({
+      hasUnicode: false,
+      hasColor: true,
+    }),
 );
 
-export let SIGINT: boolean = false;
-export let SIGKILL: boolean = false;
-const PARALLEL = (isNaN as any)(process.env['PARALLEL'])
-  ? 36
-  : parseInt(process.env['PARALLEL'] || '36');
+export const SIGINT = false;
+export let SIGKILL = false;
+const PARALLEL = (isNaN as any)(process.env["PARALLEL"]) ?
+  36 :
+  parseInt(process.env["PARALLEL"] || "36");
 
-export let topHash: string = '';
+export let topHash = "";
 export let topHeight: CassandraTypes.Long = toLong(0);
 export let topTxIndex: CassandraTypes.Long = toLong(0);
 
 const developmentSyncLength: number | undefined =
-  !process.env['DEVELOPMENT_SYNC_LENGTH'] ||
-  R.isEmpty(process.env['DEVELOPMENT_SYNC_LENGTH'])
-    ? undefined
-    : parseInt(process.env['DEVELOPMENT_SYNC_LENGTH'] as string);
+  !process.env["DEVELOPMENT_SYNC_LENGTH"] ||
+  R.isEmpty(process.env["DEVELOPMENT_SYNC_LENGTH"]) ?
+    undefined :
+    parseInt(process.env["DEVELOPMENT_SYNC_LENGTH"] as string);
 
-if (developmentSyncLength === NaN) {
-  console.error('Development sync range variable produced, illegal value NaN');
+if (isNaN(developmentSyncLength)) {
```

Comment:
this caused the test to break, the test for isNaN is true on any non-number, whereas === NaN is a check for the primitive value of NaN which is never true for strings as example.

Reply:
Noted - I'm happy with the change